up.go

```diff
@@ -794,8 +843,23 @@ var (
 	whiteOnDBlue = tcell.StyleDefault.Foreground(tcell.ColorWhite).Background(tcell.ColorNavy)
 )
 
-func drawText(region Region, style tcell.Style, text string) {
-	for x, ch := range text {
-		region.SetCell(x, 0, style, ch)
+func drawText(region Region, x int, style tcell.Style, text string) {
+	// the primary non-zero width rune
+	var mainc rune
+	// the array that follows is a possible list of combining characters to append
+	combc := make([]rune, 0)
+
+	for _, ch := range text {
+		if unicode.IsMark(ch) {
+			combc = append(combc, ch)
+		} else {
+			region.SetContent(x, 0, mainc, combc, style)
```

Comment:
@akavel

Thanks for your advice.

In fact, I was just thinking asking for help about this function. As you pointed out, if å appears, a zero-width character are printed. But because it has no width, the variable w is not incremented. As the result, the following loop sequence begin with the first byte of this line.

I think this is a little hard to understand and tend to be a cause of bug. I think your way more suitable.

Reply:
Modified drawText function as you recommended. Thanks. 

Please review https://github.com/akavel/up/pull/42/commits/243eb17938d83342342db4161cb759a1e8f2e9ba